drivers/spi/spi_silabs_eusart.c

```diff
@@ -152,6 +227,343 @@ static int spi_silabs_eusart_configure(const struct device *dev, const struct sp
 	return 0;
 }
 
+static inline void spi_silabs_eusart_pm_policy_get(const struct device *dev)
+{
+#if defined(CONFIG_SPI_SILABS_EUSART_DMA)
+	const struct spi_silabs_eusart_config *eusart_config = dev->config;
+
+	if (eusart_config->is_dma_enabled_instance)
+#endif
+	{
+		pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+	}
+	pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+	pm_policy_state_lock_get(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+	pm_device_runtime_get(dev);
+}
+
+static inline void spi_silabs_eusart_pm_policy_put(const struct device *dev)
+{
+#if defined(CONFIG_SPI_SILABS_EUSART_DMA)
+	const struct spi_silabs_eusart_config *eusart_config = dev->config;
+
+	if (eusart_config->is_dma_enabled_instance)
+#endif
+	{
+		pm_policy_state_lock_get(PM_STATE_RUNTIME_IDLE, PM_ALL_SUBSTATES);
+	}
+	pm_policy_state_lock_put(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+	pm_policy_state_lock_put(PM_STATE_STANDBY, PM_ALL_SUBSTATES);
+	pm_device_runtime_put(dev);
+}
+
+static int spi_silabs_eusart_power_management_action(const struct device *dev,
+						     enum pm_device_action action)
+{
+	int ret;
+
+	const struct spi_silabs_eusart_config *eusart_config = dev->config;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		ret = clock_control_on(eusart_config->clock_dev,
+				       (clock_control_subsys_t)&eusart_config->clock_cfg);
+		if (ret == -EALREADY) {
+			/* ignore if clock is already on */
+			ret = 0U;
+		} else if (ret < 0) {
+			return ret;
+		}
+
+		pinctrl_apply_state(eusart_config->pcfg, PINCTRL_STATE_DEFAULT);
+
+		EUSART_Enable(eusart_config->base, eusartEnable);
+
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		ret = pinctrl_apply_state(eusart_config->pcfg, PINCTRL_STATE_SLEEP);
+
+		EUSART_Enable(eusart_config->base, eusartDisable);
+		ret = clock_control_off(eusart_config->clock_dev,
+					(clock_control_subsys_t)&eusart_config->clock_cfg);
+
+		if (ret == -EALREADY) {
+			/* ignore if clock is already on */
+			ret = 0U;
+		} else if (ret < 0) {
+			return ret;
+		}
+
+		break;
+	default:
+		ret = -ENOTSUP;
+	}
+
+	return ret;
+}
+
+#if defined(CONFIG_SPI_SILABS_EUSART_DMA)
+static inline void spi_silabs_eusart_clear_txrx_fifos(EUSART_TypeDef *eusart)
+{
+	eusart->CMD_SET = EUSART_CMD_CLEARTX;
+
+	while (EUSART_StatusGet(eusart) & EUSART_STATUS_RXFL) {
+		(void)eusart->RXDATA;
```

Comment:
Even if Cortex-M is not tricky, I suggest to use accessors to access to hardware registers:
```
sys_read32(&eusart->RXDATA);
```
These function, add the necessary memory barriers (on compiler and on CPU sides).